app/helpers/browse_helper.rb

```diff
@@ -1,21 +1,10 @@
 module BrowseHelper
-  def display_popular_links_for_slug?(slug)
-    popular_links_data(slug).present?
-  end
-
-  def popular_links_data(slug)
-    if I18n.exists?("browse.popular_links.#{slug}", :en)
-      I18n.t("browse.popular_links.#{slug}")
-    end
-  end
-
-  def popular_links_for_slug(slug)
-    links = popular_links_data(slug)
-    count = links.length
-    links.map.with_index(1) do |link, index|
+  def with_tracking_data(popular_content)
+    count = popular_content.length
+    popular_content.map.with_index(1) do |page, index|
```

Comment:
I think `popular_content` is a list of links, not a page object? So perhaps the variable name should be left as link?

That'll also match the naming in the view and remove a bit of confusion.